### PR TITLE
【Zero-Dim】Support Zero dim for embedding and one-hot

### PR DESCRIPTION
**Pull request by seemingwang**
### PR types
Others
### PR changes
Others
### Describe
support zero-dimension input for embedding and one-hot api-

### CODE REVIEW

File: paddle/phi/infermeta/unary.cc [hunk 1]
```diff
@@ -2230,10 +2231,11 @@ void OneHotInferMeta(const MetaTensor& x,
                      const Scalar& depth_t,
                      MetaTensor* out) {
   auto x_dims = x.dims();
-  PADDLE_ENFORCE_GE(
-      x_dims.size(),
-      1,
-      phi::errors::InvalidArgument("Rank of Input(X) should be at least 1."));
+  // PADDLE_ENFORCE_GE(
```
Comment (zhwesky2010) [inline paddle/phi/infermeta/unary.cc]: 这个改成>=0吧

File: paddle/phi/infermeta/unary.cc [hunk 2]
```diff
@@ -2214,10 +2214,11 @@ void OneHotRawInferMeta(const MetaTensor& x,
                         bool allow_out_of_range,
                         MetaTensor* out) {
   auto x_dims = x.dims();
-  PADDLE_ENFORCE_GE(
-      x_dims.size(),
-      1,
-      phi::errors::InvalidArgument("Rank of Input(X) should be at least 1."));
+  // PADDLE_ENFORCE_GE(
```
Comment (zhwesky2010) [inline paddle/phi/infermeta/unary.cc]: 这个改成>=0吧

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 3]
```diff
@@ -1213,6 +1231,36 @@ def test_zeros_and_zeros_like(self):
...
+        ids = paddle.full(shape=[], fill_value=1, dtype='int64')
+        w0 = paddle.arange(3, 9).reshape((3, 2)).astype(paddle.float32)
+        w = paddle.to_tensor(w0, stop_gradient=False)
+        emb = paddle.nn.functional.embedding(
+            x=ids, weight=w, sparse=True, name="embedding"
+        )
+
+        prog = paddle.static.default_main_program()
+        res = self.exe.run(prog, fetch_list=[emb])
+        self.assertEqual(res[0].shape, (2,))
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1244]: 可以加个测试下输出值

**Review by zhwesky2010 — COMMENTED** [review]
customdevice 与主repo case一致

**Review by zhwesky2010 — DISMISSED** [review]
LGTM

**Review by zhwesky2010 — APPROVED** [review]
